src/test_mode/src/wrapper.rs

```diff
@@ -41,60 +63,159 @@ pub fn wrapper_init(){
+//  -get sensor data
+//  -check endscript property & exit application as indicated
+//  -update the sensor interface with gyro data from jsbsim
+//                        
+///////////////////////////////////////////////////////////////////////////////
 pub fn get_from_jsbsim()->(f32, f32, f32){
+
+    let runresult: bool;
+    
     unsafe{
-        return (gyro_x, gyro_y, gyro_z);
+        //iterate jsbsim & verify
+        runresult = fdm_run(fdm);
+        if runresult==false{
+            panic!("jsbsim did not iterate properly");
+        }
+        
+        //get sensor data
+        let gx: f64;
+        let gy: f64 = 0.0;
+        let gz: f64 = 0.0;
+        
+        let property_gyro = std::ffi::CString::new("testmode/gyro").unwrap();
+        gx = fdm_get_property_double(fdm, property_gyro.as_ptr());
+        
+        //temporary quit mechanism
+        let endscript: f64;
+        let property_endscript = std::ffi::CString::new("testmode/endscript").unwrap();
+        endscript = fdm_get_property_double(fdm, property_endscript.as_ptr());
+        
+        
+        if endscript > 0.0{
+            println!("endscript:\t{}", endscript);
+            panic!("temporary quit mechanism");
+        }
+                
+        //return gyro readings
+        return (gx as f32, gy as f32, gz as f32);
     }
 }
 
-//development:  jsbsim destructor
+
+///////////////////////////////////////////////////////////////////////////////
+//
+//  Function name:      wrapper_close
+//
+//  Purpose:            close the FDM and reset fdm to null (not implemented)
+//
+//  Methodology:
+//  -close FDM
+//  -set fdm to null
+//                        
+///////////////////////////////////////////////////////////////////////////////
 pub fn wrapper_close(){
-    println!("binder close");
+    //fdm_close(fdm);
+    //fdm = 0 as *mut FDM;
 }
 
+
+///////////////////////////////////////////////////////////////////////////////
+//
+//  Section name:       extern block
+//
+//  Purpose:            provide rust-based front end for the c abi defined
+//                          in wrapper.h and wrapper.cpp
+//
+//  Methodology:
+//  -provide linkage to JSBSim as a shared object or dynamic link library
+//  -provide linkage to wrapper.h & wrapper.cpp as a static library
+//  -provide a basic set of functions to access JSBSim via the c abi defined
+//      in wrapper.h & wrapper.cpp
+//
+//  Notes:
+//  -the functions in the extern block must parallel the c headers in wrapper.h
+//                        
+///////////////////////////////////////////////////////////////////////////////
 #[link(name = "stdc++")]
 #[link(name = "JSBSim")]
 #[link(name = "wrapper", kind = "static")]
-//rust wrapper definitions using c abi
+
 extern "C" {
     //jsbsim constructor
     fn fdm_create()->*mut FDM;
 
     //jsbsim destructor
-    fn fdm_close(fdm: *mut FDM);
+    //fn fdm_close(fdm: *mut FDM);      //exit is not currently implemented
```

Comment:
Why is this even here? I don't see a reason for it since we don't have a way to end the simulation.